src/Datagrid/ListMapper.php

```diff
@@ -77,6 +77,17 @@ public function add($name, $type = null, array $fieldDescriptionOptions = [])
             $fieldDescriptionOptions['virtual_field'] = true;
         }
 
+        if (\array_key_exists('identifier', $fieldDescriptionOptions) && !\is_bool($fieldDescriptionOptions['identifier'])) {
+            @trigger_error(
+                'Passing a non boolean value for the "identifier" option is deprecated since sonata-project/admin-bundle 3.x and will throw an exception in 4.0.',
+                E_USER_DEPRECATED
+            );
+
+            $fieldDescriptionOptions['identifier'] = (bool) $fieldDescriptionOptions['identifier'];
+            // NEXT_MAJOR: Remove the previous 6 lines and use commented line below it instead
```

Comment:
I thought that these `NEXT_MAJOR` comments are resolved while merging against `master` branch.

Reply:
Nevermind. I just saw the following paragraph at `CONTRIBUTING.md`:
>If you have a non-BC PR to propose, please try to create a related BC PR first. This BC PR should mark every piece of code that needs to be removed / uncommented / reworked in the corresponding non-BC PR with the following marker comment : NEXT_MAJOR. When the BC PR is merged in the stable branch, wait for the stable branch to be merged in the unstable branch, and then work on your non-BC PR.